truss/contexts/local_loader/truss_file_syncer.py

```diff
@@ -1,27 +1,84 @@
+import logging
 from pathlib import Path
 from threading import Thread
+from typing import List, Optional
 
-from truss.util.path import copy_file_path
+from truss.constants import CONFIG_FILE
+from truss.patch.calc_patch import calc_config_patches
+from truss.patch.local_truss_patch_applier import LocalTrussPatchApplier
+from truss.templates.control.control.helpers.types import (
+    Action,
+    ModelCodePatch,
+    Patch,
+    PatchType,
+)
+from truss.truss_config import TrussConfig
+from truss.truss_spec import TrussSpec
+from truss.util.path import file_content, is_ignored, load_trussignore_patterns
 from watchfiles import Change, watch
 
+OP_2_ACTION = {
+    Change.added: Action.ADD,
+    Change.deleted: Action.REMOVE,
+    Change.modified: Action.UPDATE,
+}
+
+
+class TrussPatchEmitter:
+    def __init__(
+        self,
+        truss_dir: Path,
+        logger: logging.Logger,
+    ) -> None:
+        self._truss_dir = truss_dir
+        self._config = TrussConfig.from_yaml(self._truss_dir / CONFIG_FILE)
+        self._logger = logger
+
+    def __call__(self, op, path: Path) -> Optional[List[Patch]]:
+        truss_spec = TrussSpec(self._truss_dir)
+        model_module_path = str(
+            truss_spec.model_module_dir.relative_to(self._truss_dir)
+        )
+        if str(path).startswith(model_module_path):
+            rel_path = str(path.relative_to(model_module_path))
+            return [
+                Patch(
+                    type=PatchType.MODEL_CODE,
+                    body=ModelCodePatch(
+                        OP_2_ACTION[op],
+                        rel_path,
+                        file_content(self._truss_dir / path),
```

Comment:
Actually pathlib make is really easy to read content, path.read_text(), so we don't need file_content helper function.